src/commands/node.mjs

```diff
@@ -2314,8 +2355,766 @@ export class NodeCommand extends BaseCommand {
+        }
+      },
+      {
+        title: 'Prepare grpc service endpoints',
+        task: (ctx, task) => {
+          const config = /** @type {NodeUpdateConfigClass} **/ ctx.config
+          let endpoints = []
+
+          if (!config.grpcEndpoints) {
+            if (config.endpointType !== constants.ENDPOINT_TYPE_FQDN) {
+              throw new FullstackTestingError(`--grpc-endpoints must be set if --endpoint-type is: ${constants.ENDPOINT_TYPE_IP}`)
+            }
+
+            endpoints = [
+              `${Templates.renderFullyQualifiedNetworkSvcName(config.namespace, config.nodeId)}:${constants.HEDERA_NODE_EXTERNAL_GOSSIP_PORT}`
+            ]
+          } else {
+            endpoints = helpers.splitFlagInput(config.grpcEndpoints)
+          }
+
+          ctx.grpcServiceEndpoints = this.prepareEndpoints(config.endpointType, endpoints, constants.HEDERA_NODE_EXTERNAL_GOSSIP_PORT)
+        }
+      },
+      {
+        title: 'Load node admin key',
+        task: async (ctx, task) => {
+          const config = /** @type {NodeUpdateConfigClass} **/ ctx.config
+          config.adminKey = PrivateKey.fromStringED25519(constants.GENESIS_KEY)
+        }
+      },
+      {
+        title: 'Prepare upgrade zip file for node upgrade process',
+        task: async (ctx, task) => {
+          const config = /** @type {NodeUpdateConfigClass} **/ ctx.config
+          ctx.upgradeZipFile = await this.prepareUpgradeZip(config.stagingDir)
+          ctx.upgradeZipHash = await this.uploadUpgradeZip(ctx.upgradeZipFile, config.nodeClient)
+        }
+      },
+      {
+        title: 'Check existing nodes staked amount',
+        task: async (ctx, task) => {
+          const config = /** @type {NodeUpdateConfigClass} **/ ctx.config
+          await sleep(60000)
+          const accountMap = getNodeAccountMap(config.existingNodeIds)
+          for (const nodeId of config.existingNodeIds) {
+            const accountId = accountMap.get(nodeId)
+            await this.accountManager.transferAmount(constants.TREASURY_ACCOUNT_ID, accountId, 1)
+          }
+        }
+      },
+      {
+        title: 'Send node update transaction',
+        task: async (ctx, task) => {
+          const config = /** @type {NodeUpdateConfigClass} **/ ctx.config
+
+          const nodeId = Templates.nodeNumberFromNodeId(config.nodeId) - 1
+          self.logger.info(`nodeId: ${nodeId}`)
+          self.logger.info(`config.newAccountNumber: ${config.newAccountNumber}`)
+
+          self.logger.info(`ctx.signingCertDer: ${ctx.signingCertDer}`)
+          self.logger.info(`ctx.tlsCertHash: ${ctx.tlsCertHash}`)
+          try {
+            const nodeUpdateTx = await new NodeUpdateTransaction()
+              .setNodeId(nodeId)
+              .setGossipEndpoints(ctx.gossipEndpoints)
+              .setServiceEndpoints(ctx.grpcServiceEndpoints)
+              .setGossipCaCertificate(ctx.signingCertDer)
+              .setCertificateHash(ctx.tlsCertHash)
```

Comment:
why set it and then overlay it.  it would be better to only set it once with whatever the correct value is.